ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_volume.py

```diff
@@ -822,10 +827,16 @@ def main():
                     limit_iops=limit_iops,
                     limit_mbps=limit_mbps)
             else:
-                # if we are here it means volume is already present. hence issue update call
+                if volcoll is not None and volcoll == "":
+                    # volcoll can be an empty string when user wants to disassociate volume from volcoll
+                    disassociate = True
+                else:
+                    disassociate = False
+
                 return_status, changed, msg, changed_attrs_dict = update_volume(
                     client_obj,
                     vol_resp,
+                    disassociate,
```

Comment:
As @rgcostea mentioned above, you can pass the `volcoll` as kwargs and check if its empty string and perform disassociate.

`volcoll=volcoll`